### PR TITLE
[MINOR][BUILD] Fix lint-java breaks in Java

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to fix the lint-breaks as below:

```
[ERROR] src/test/java/org/apache/spark/network/TransportResponseHandlerSuite.java:[29,8] (imports) UnusedImports: Unused import - org.apache.spark.network.buffer.ManagedBuffer.
[ERROR] src/main/java/org/apache/spark/unsafe/types/UTF8String.java:[156,10] (modifier) ModifierOrder: '@Nonnull' annotation modifier does not precede non-annotation modifiers.
[ERROR] src/main/java/org/apache/spark/SparkFirehoseListener.java:[122] (sizes) LineLength: Line is longer than 100 characters (found 105).
[ERROR] src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java:[164,78] (coding) OneStatementPerLine: Only one statement per line allowed.
[ERROR] src/test/java/test/org/apache/spark/JavaAPISuite.java:[1157] (sizes) LineLength: Line is longer than 100 characters (found 121).
[ERROR] src/test/java/org/apache/spark/streaming/JavaMapWithStateSuite.java:[149] (sizes) LineLength: Line is longer than 100 characters (found 113).
[ERROR] src/test/java/test/org/apache/spark/streaming/Java8APISuite.java:[146] (sizes) LineLength: Line is longer than 100 characters (found 122).
[ERROR] src/test/java/test/org/apache/spark/streaming/JavaAPISuite.java:[32,8] (imports) UnusedImports: Unused import - org.apache.spark.streaming.Time.
[ERROR] src/test/java/test/org/apache/spark/streaming/JavaAPISuite.java:[611] (sizes) LineLength: Line is longer than 100 characters (found 101).
[ERROR] src/test/java/test/org/apache/spark/streaming/JavaAPISuite.java:[1317] (sizes) LineLength: Line is longer than 100 characters (found 102).
[ERROR] src/test/java/test/org/apache/spark/sql/JavaDatasetAggregatorSuite.java:[91] (sizes) LineLength: Line is longer than 100 characters (found 102).
[ERROR] src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java:[113] (sizes) LineLength: Line is longer than 100 characters (found 101).
[ERROR] src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java:[164] (sizes) LineLength: Line is longer than 100 characters (found 110).
[ERROR] src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java:[212] (sizes) LineLength: Line is longer than 100 characters (found 114).
[ERROR] src/test/java/org/apache/spark/mllib/tree/JavaDecisionTreeSuite.java:[36] (sizes) LineLength: Line is longer than 100 characters (found 101).
[ERROR] src/main/java/org/apache/spark/examples/streaming/JavaKinesisWordCountASL.java:[26,8] (imports) UnusedImports: Unused import - com.amazonaws.regions.RegionUtils.
[ERROR] src/test/java/org/apache/spark/streaming/kinesis/JavaKinesisStreamSuite.java:[20,8] (imports) UnusedImports: Unused import - com.amazonaws.regions.RegionUtils.
[ERROR] src/test/java/org/apache/spark/streaming/kinesis/JavaKinesisStreamSuite.java:[94] (sizes) LineLength: Line is longer than 100 characters (found 103).
[ERROR] src/main/java/org/apache/spark/examples/ml/JavaTokenizerExample.java:[30,8] (imports) UnusedImports: Unused import - org.apache.spark.sql.api.java.UDF1.
[ERROR] src/main/java/org/apache/spark/examples/ml/JavaTokenizerExample.java:[72] (sizes) LineLength: Line is longer than 100 characters (found 104).
[ERROR] src/main/java/org/apache/spark/examples/mllib/JavaRankingMetricsExample.java:[121] (sizes) LineLength: Line is longer than 100 characters (found 101).
[ERROR] src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java:[28,8] (imports) UnusedImports: Unused import - org.apache.spark.api.java.JavaRDD.
[ERROR] src/main/java/org/apache/spark/examples/sql/JavaSQLDataSourceExample.java:[29,8] (imports) UnusedImports: Unused import - org.apache.spark.api.java.JavaSparkContext.
```

## How was this patch tested?

Manually via 

```bash
./dev/lint-java
```

### CODE REVIEW

File: common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ public void writeTo(ByteBuffer buffer) {
    *
    * Unlike getBytes this will not create a copy the array if this is a slice.
    */
-  public @Nonnull ByteBuffer getByteBuffer() {
+  @Nonnull public ByteBuffer getByteBuffer() {
```
Comment (HyukjinKwon) [inline common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java]: This seems actually a correct order but there is a bug in checkstyle. Please see `https://github.com/checkstyle/checkstyle/issues/903`. I am willing to disable `ModifierOrder` for this method if anyone thinks so.

**Comment by HyukjinKwon** [comment]
I haven't checked the tests. Maybe, let me wait for the test results before cc'ing someone.

**Comment by HyukjinKwon** [comment]
retest this please

File: common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java (hunk 1 above)
Comment (srowen) [inline common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java]: The compiler accepts it but I agree that annotations conventionally come before modifiers. They usually come on the line above.

**Comment by srowen** [comment]
Thank you, yes just last night I put together some changes to fix this too, because I added some problems when I changed to Java 8 syntax. Maybe you can apply https://github.com/srowen/spark/commit/fcc66a0a79399b81c08c17f93073a9635de90a95.patch to your branch to see if I got anything else?

File: core/src/main/java/org/apache/spark/SparkFirehoseListener.java [hunk 2]
```diff
@@ -34,12 +34,12 @@ public void onEvent(SparkListenerEvent event) { }
 
     @Override
     public final void onStageCompleted(SparkListenerStageCompleted stageCompleted) {
-        onEvent(stageCompleted);
+      onEvent(stageCompleted);
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/SparkFirehoseListener.java]: Yeah, the whole file is mis-indented. The first indent is still 4 spaces. I think you could fix that too if you like

Comment (HyukjinKwon) [inline core/src/main/java/org/apache/spark/SparkFirehoseListener.java]: Ah, sure. Thank you. Let me address the comments and compare this to your work too.

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thank you @srowen.

**Pull request closed without merging**